src/components/Vote/Vote.tsx

```diff
@@ -0,0 +1,68 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import React from 'react';
+
+import { ComponentPropsWithoutRef } from 'react';
+
+import { TbTriangle, TbTriangleFilled } from 'react-icons/tb';
+
+export interface VoteProps extends ComponentPropsWithoutRef<'div'> {
+  votes: number;
+  voted?: 'VOTE' | 'UNVOTE';
+  onVoteUp: () => void;
+  onVoteDown: () => void;
+  customClass?: string;
+}
+
+export function Vote(props: VoteProps) {
+  const { votes, voted, onVoteUp, onVoteDown, customClass, ...rest } = props;
+
+  function onClickItem(arg: 1 | -1) {
+    switch (voted) {
+      case 'VOTE': {
+        arg === 1 ? onVoteDown() : (onVoteDown(), onVoteDown());
```

Comment:
🤔 do we need the function twice!?

Reply:
> thinking do we need the function twice!?

I will simplify the logic...
As it's the backend that has to say whether the user voted `up` or `down`. I'll just replace the 2 voting functions (`onVoteUp` & `onVoteDown`) with `onVote`.

```typescript
<div
  data-testid="vote-up"
  className="cursor-pointer"
  onClick={onVote}
>
  {voted === 'VOTE' ? (
    <TbTriangleFilled data-testid="up-filled" />
  ) : (
    <TbTriangle data-testid="up-not-filled" />
  )}
</div>
```

Reply:
Ohhh much better and beautiful! Great @maxime-carabina !!! 👏